run.go

```diff
@@ -206,3 +213,23 @@ func proxyDirector(super func(*http.Request)) func(*http.Request) {
 		req.Host = req.URL.Host
 	}
 }
+
+// HTTPProtoErrorHandler is a function that overrides standard Handler.
+// Enhance the error message so that users can understand better the API behaviour.
+// Detect the problem data type and return a message that is relevant.
+func HTTPProtoErrorHandler(ctx context.Context, mux *runtime.ServeMux, marshaler runtime.Marshaler, w http.ResponseWriter, request *http.Request, err error) {
+	if err != nil {
+		var regex = regexp.MustCompile(`cannot unmarshal (\w+)\s`)
+		st, _ := status.FromError(err)
+		message := st.Message()
+		match := regex.FindStringSubmatch(message)
+		if match != nil {
+			s := status.New(codes.InvalidArgument, "You can't pass "+match[1]+" type data to the datadictionary API.")
```

Comment:
`fmt.Sprintf` style should be preferred over concatenation with `+`. In this case, there's a `status.Newf` specifically for the purpose.

I'd also suggest pulling out the expected type of the data since that should also be in the original message. Then our error message doesn't lose that information.

```suggestion
			s := status.Newf(codes.InvalidArgument, "%s type data received for %s type field", match[1], match[2])
```